hbase-common/src/main/java/org/apache/hadoop/hbase/nio/ByteBuff.java

```diff
@@ -78,6 +79,23 @@ public boolean release() {
     return refCnt.release();
   }
 
+  public RefCnt getRefCnt() {
+    return this.refCnt;
+  }
+
+  /**
+   * BucketEntry use this to share refCnt with ByteBuff, so make the method public here,
+   * the upstream should not use this public method in other place, or the previous recycler
+   * will be lost.
+   */
+  public void shareRefCnt(RefCnt refCnt, boolean replace) {
```

Comment:
In case of File based BC, now we make it to read the cached data into the pooled BBs.  Every read RPC will acquire own BBs and read into. There is ideally no sharing of BBs across the readers happen at all.. But seems here we try share the ref count of the BC (File based here) with the RPCs.  Little strange to digest.

Reply:
Sorry, there missing some conversations with @openinx 
We do this in order not to violate the LRU, when BucketCache#freeEntireBuckets is executed